### ISSUE TITLE
🚀 Feature: [Catalog frontend] Possibility to remove on of the action icon edit on catalog panel

### ISSUE THREAD

**Issue by RaiffMello1** [backstage/backstage#28072]
### 🔖 Feature description

As a developer I would like to have the possibility to remove the **_edit action icon_** .
![Screenshot 2024-12-10 at 11 35 20](https://github.com/user-attachments/assets/3d7a47d9-1672-4e6c-8ed8-8e77b9df0a82)

### 🎤 Context

In the context of my company we dont need it **_EDIT icon_** and we want to know the possibility to remove this icon for a cleaner interface.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by awanlin** [comment]
Hi @RaiffMello1, this should help, be sure to read the text below the note as it will help for your use case: https://backstage.io/docs/features/software-catalog/catalog-customization#customize-actions

**Comment by RaiffMello1** [comment]
Hi @awanlin reading the docs I saw the possibility to ADD an action for one of the action icons as in the link you suggested. But what I want to achieve here its to have the possibility to REMOVE the icon itself.
What do you think

**Comment by awanlin** [comment]
Did you read the test below the note that I mentioned?

**Comment by freben** [comment]
Yeah that's slightly awkward - you'd need to wrap your entire catalog table in a helper component that can use hooks, and then generates the actions that it then passes down into the table.

Slight side note also that this action is disabled [if the `backstage.io/edit-url` annotation isn't present](https://github.com/backstage/backstage/blob/703048420a68333017f375c23b5213c2b969f3a5/plugins/catalog/src/components/CatalogTable/CatalogTable.tsx#L153). So maybe ensuring that it's always deleted (like in a custom processor or so) would entirely disable this family of functionality. But the icon would be visible and greyed out still.
